script/upgrades/MU08/MU08.sol

```diff
@@ -126,126 +125,69 @@ contract MU08 is IMentoUpgrade, GovernanceScript {
     proposal_transferTokenOwnership();
     proposal_transferMentoV2Ownership();
     proposal_transferMentoV1Ownership();
+    proposal_transferGovFactoryOwnership();
 
     return transactions;
   }
 
   function proposal_transferTokenOwnership() public {
     address[] memory tokenProxies = Arrays.addresses(cUSDProxy, cEURProxy, cBRLProxy, eXOFProxy, cKESProxy);
-    //address[] memory tokenProxies = Arrays.addresses(cUSDProxy, cEURProxy, cBRLProxy, eXOFProxy, cKESProxy, POSProxy);
     for (uint i = 0; i < tokenProxies.length; i++) {
-      address proxyOwner = IOwnableLite(tokenProxies[i]).owner();
-      // Transfer ownership of the token proxies
-      if (proxyOwner != timelockProxy) {
-        transactions.push(
-          ICeloGovernance.Transaction({
-            value: 0,
-            destination: tokenProxies[i],
-            data: abi.encodeWithSelector(IOwnableLite(0).transferOwnership.selector, timelockProxy)
-          })
-        );
-      }
+      transferOwnership(tokenProxies[i]);
     }
-    // Transfer ownership of the StableTokenV2 implemntation used by all proxies
-    address stableTokenImplementation = IProxyLite(cUSDProxy)._getImplementation();
-    address implementationOwner = IOwnableLite(stableTokenImplementation).owner();
 
-    if (implementationOwner != timelockProxy) {
-      require(implementationOwner == celoGovernance, "StableTokenV2 implementation owner is not Celo governance");
-      transactions.push(
-        ICeloGovernance.Transaction({
-          value: 0,
-          destination: stableTokenImplementation,
-          data: abi.encodeWithSelector(IOwnableLite(0).transferOwnership.selector, timelockProxy)
-        })
-      );
-    }
+    // Transfer ownership of the cUSD implementation all the token proxies
+    // are pointing to the same StableTokenV2 implementation
```

Comment:
can we have a check/assertion for this before doing the transfer?

Reply:
Do you mean a check verifying that all proxies point to the same implementation? 
The checks kind of verify that and would complain if any of the implementations are not owned by the TimelockProxy after the proposal